Dashboard domain card UI

ravishanker opened the pull request:
Fixes #18178 

<img width=320 src="https://user-images.githubusercontent.com/990349/230291552-4864c51c-e672-427f-8b38-c136efe48405.png" />

To test:

1. Launch Jetpack/WordPress app
2. Go to **App Settings** (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
3. Tap **Debug Settings**
4. Find **dashboard_card_domain** under Remote features as shown in the image above
5. Verify **domain card** appears as shown above on dashboard
6. Tap on overflow (three dots) menu
7. **Verify** it pops up menu with **Hide this** option
8. Tap on Hide this
9. **Verify**, the card gets hidden 

>** Note **
> The 🌐 domain icon needs to be updated, waiting for export.  Will update before merging or in the next PR

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and manual tests

3. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Add Dashboard Domain Card

Adds data class DashboardDomainCard

ravishanker committed:
Add Dashboard Domain Card Builder

Adds
- DashboardDomainCardBuilder
- ViewHolder
- layout
- string resources
- menu

ravishanker committed:
Update builder, adapter, and tracker

ravishanker committed:
Update CardsBuilderTest.kt

ravishanker committed:
Update DashboardDomainCardBuilderTest.kt

fix detekt line length issue

ravishanker committed:
Merge branch 'trunk' into Dashboard-Domain-Card-UI

staskus commented:
@ravishanker, thanks!

✅ The UI looks good!

The presentation logic seems to be *mostly* working. The logic for checking existing domains is still needed:

```
// !hasSiteDomains(siteModel) && // this may need a separate api call!
```

ravishanker commented:
> The presentation logic seems to be _mostly_ working. The logic for checking existing domains is still needed:
> 
> ```
> // !hasSiteDomains(siteModel) && // this may need a separate api call!
> ```
Thank you, coming in the next PR

irfano commented on WordPress/src/main/res/layout/dashboard_card_domain.xml:
```diff
@@ -0,0 +1,71 @@
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:background="?attr/selectableItemBackgroundBorderless"
+                android:clickable="true"
+                android:contentDescription="@string/content_description_more"
+                android:focusable="true"
```
❓ Are `android:clickable="true"` and `android:focusable="true"` necessary here? I removed them, and the more button is still focusable and clickable.

irfano commented on WordPress/src/main/res/layout/dashboard_card_domain.xml:
```diff
@@ -0,0 +1,71 @@
+                android:id="@+id/dashboard_card_domain_sub_title"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:paddingStart="@dimen/margin_medium"
+                android:paddingEnd="@dimen/margin_medium"
+                android:paddingBottom="@dimen/margin_medium"
+                android:paddingTop="@dimen/margin_medium"
```
Please consider adding using `android:padding="@dimen/margin_medium"` instead of setting four different paddings separately.

irfano commented on WordPress/src/main/res/layout/dashboard_card_domain.xml:
```diff
@@ -0,0 +1,71 @@
+                android:layout_height="wrap_content"
+                android:text="@string/dashboard_card_domain_title"
+                android:textAlignment="viewStart"
+                android:textAppearance="@style/blaze_title"
```
I think it would be nice to rename the Blaze styles that are used in both Blaze and domain cards. Using a more common name could be better. Perhaps `simple_domain_card_styles.xml`, `simple_domain_card_title`, or any other name that you think would be more appropriate.

irfano commented on WordPress/src/main/res/layout/dashboard_card_domain.xml:
```diff
@@ -0,0 +1,71 @@
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                app:tint="?attr/wpColorOnSurfaceMedium"
+                tools:ignore="TouchTargetSizeCheck" />
```
I don't know why `TouchTargetSizeCheck` was added on `promote_with_blaze_card`. I removed it, ran checks, and couldn't find any error. I think we can remove it if it doesn't suppress any warning.

irfano commented on WordPress/src/main/res/layout/dashboard_card_domain.xml:
```diff
@@ -0,0 +1,71 @@
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <ImageView
+                android:id="@+id/dashboard_card_domain_icon"
+                android:src="@drawable/ic_domains_white_24dp"
+                android:background="@color/jetpack_green"
+                android:contentDescription="@string/content_description_more"
+                android:clickable="true"
+                android:focusable="true"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                app:layout_constraintTop_toBottomOf="@+id/dashboard_card_domain_title"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                tools:ignore="TouchTargetSizeCheck"/>
```
We can remove `contentDescription`, `clickable`, `focusable`, `TouchTargetSizeCheck` from this view. Since it's not important for accessiiblity, we can also add `android:importantForAccessibility="no"`.
My other comments are ignorable, only this is important because when I navigate between focusable objects while testing, this icon is being focused unnecessarily.

irfano reviewed (changes requested):
It works great, passed all of my tests. Good job!
I've added minor comments and feedback. Some of them might be pointless if we want to update the UI of the card in this PR.

ravishanker committed:
Update layout and styles

Updated layout and removed redundant properties

ravishanker commented:
> It works great, passed all of my tests. Good job! I've added minor comments and feedback. Some of them might be pointless if we want to update the UI of the card in this PR.

I've updated the layout with the suggestions.  It might all change, if the design changes!

irfano reviewed (approved):
Thank you for considering my comments, @ravishanker! I'm approving it, although I have an extra minor suggestion. We can remove `blaze_styles.xml` and use `dashboard_card_styles` in `promote_with_blaze_card.xml`. Since they are exactly the same.

irfano commented:
@ravishanker, We can also add release notes in this PR.
```
* [**] [Jetpack-only] Added a dashboard card for purchasing domains. [https://github.com/wordpress-mobile/WordPress-Android/pull/18240]
```

ravishanker committed:
Update RELEASE-NOTES.txt

ravishanker committed:
Merge branch 'trunk' into Dashboard-Domain-Card-UI

ravishanker commented:
> Thank you for considering my comments, @ravishanker! I'm approving it, although I have an extra minor suggestion. We can remove `blaze_styles.xml` and use `dashboard_card_styles` in `promote_with_blaze_card.xml`. Since they are exactly the same.

`blaze_styles.xml` may need to be updated for the next phase of the project, so better leave it for now.
